File: Contests/PSTU/30 aug 24/d.c
#include <stdio.h>
int main() {
    long int n;
    int flag;
    scanf("%ld", &n);
        if (n % 2 == 0)
        {
            flag = 0;
        }
        while (n > 1) {
                n = n - 1;
                if (flag == 1) {
                    flag = 0;
                }
                else {
                    flag = 1;
                }
            if (n == 1) {
                break;
            }
        }
        if (flag == 1) {
            printf("Mahmoud\n");
        }
        else {
            printf("Bashar\n");
        }
}

File: Contests/PSTU/30 aug 24/d.java
import java.util.Scanner;

public class d {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int flag = 0;
        int n;
        n = input.nextInt();
        while (n > 1) {
            if (n % 2 == 0) {
                n = n - (n/2);
            }
            else {
                n = n - 1;
            }
            if (n == 1) {
                break;
            }
        }
        if (flag == 0) {
            System.out.println("Mahmoud");
        }
        else {
            System.out.println("Bashar");
        }
        input.close();
    }
}
